pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -491,14 +491,21 @@ func (c *clustersReconciler) runGC(ctx context.Context) {
 			log.V(2).Info("Garbage Collector Stopped")
 			return
 		case <-time.After(c.gcInterval):
-			log.V(4).Info("Run Garbage Collection for Lost Remote Workloads")
-			for clusterName, rc := range c.remoteClients {
-				rc.runGC(ctrl.LoggerInto(ctx, log.WithValues("multiKueueCluster", clusterName)))
-			}
+			c.runGC(ctx)
 		}
 	}
 }
 
+func (c *clustersReconciler) runGC(ctx context.Context) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
```

Comment:
Should we lock while `rc.runGC` runs? Or should we just lock the access to `c.remoteClients`?

Reply:
Yes, we're locking `c.remoteClients`.

Reply:
I know you are locking. My question is whether we also need to lock while `rc.runGC` runs.
Instead, we could get a copy of the slice.

Reply:
Yes, you right it's enough to lock the access just for `c.remoteClients`. The conflict with removing remoteClient from map. So we can copy slice of `remoteClients`.